### PR TITLE
WIP: [CasADi] add patches for Windows

### PR DESCRIPTION
**Pull request by odow**
Follow-up the windows problems in https://github.com/JuliaPackaging/Yggdrasil/pull/5048.

### CODE REVIEW

File: C/CasADi/build_tarballs.jl [hunk 1]
```diff
@@ -22,6 +22,11 @@ cd build
 export CXXFLAGS="-fPIC -std=c++11"
 export CFLAGS="${CFLAGS} -fPIC"
 
+if [[ "$target" == *-mingw* ]]; then
+    atomic_patch -p1 ${WORKSPACE}/srcdir/patches/windows.patch
+    export CXX=clang++
```
Comment (giordano) [inline C/CasADi/build_tarballs.jl:27]: No, you should do https://docs.binarybuilder.org/stable/build_tips/#Using-GCC-on-macOS-and-FreeBSD

File: C/CasADi/build_tarballs.jl [hunk 2]
```diff
@@ -16,14 +16,18 @@ script = raw"""
...
+
 mkdir -p build
 cd build
 
 export CXXFLAGS="-fPIC -std=c++11"
 export CFLAGS="${CFLAGS} -fPIC"
 
 cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
-    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN%.*}_clang.cmake \
```
Comment (odow) [inline C/CasADi/build_tarballs.jl:30]: Do you mean something like this?

Comment (odow) [reply C/CasADi/build_tarballs.jl:30]: I sometimes feel like you carry around a tome of incantations for all the weird things that can happen. 

It's amazing how much work has gone into all of this to make it work.

**Comment by odow** [comment]
I guess I need to try your branch? https://github.com/JuliaPackaging/BinaryBuilderBase.jl/issues/248

**Comment by giordano** [comment]
Could try, but even with that one I had some other issues, so I'm not really sure that's enough. It really feels like we should do better in setting up clang for Windows, but I don't understand why the gcc toolchain flag is ignored

**Comment by odow** [comment]
> It really feels like we should do better in setting up clang for Windows

Hahah. I get the hint. I'll take a look.

**Comment by odow** [comment]
We didn't end up needing this.

**Pull request closed without merging**